File: easy-adapter/src/main/java/com/lxj/easyadapter/ItemViewDelegate.java
package com.lxj.easyadapter;


import android.support.annotation.NonNull;

/**
 * Created by zhy on 16/6/22.
 */
public interface ItemViewDelegate<T>
{

    int getLayoutId();

    boolean isForViewType(@NonNull T item, int position);

    void convert(@NonNull ViewHolder holder, @NonNull T t, int position);

}

File: easy-adapter/src/main/java/com/lxj/easyadapter/ItemViewDelegate.kt
package com.lxj.easyadapter

/**
 * Created by zhy on 16/6/22.
 */
interface ItemViewDelegate<T> {

    val layoutId: Int

    fun isForViewType(item: T, position: Int): Boolean

    fun bind(holder: ViewHolder, t: T, position: Int)

}
